tools/clang_tidy/lib/clang_tidy.dart

```diff
@@ -171,23 +189,86 @@ class ClangTidy {
+        final Iterable<String> filePaths = set.map((Command e) => e.filePath);
+        if (!filePaths.contains(command.filePath)) {
+          return false;
+        }
+      }
+      return true;
+    }
+    for (final Command command in items) {
+      if (allSetsContain(command)) {
+        yield _SetStatusCommand(_SetStatus.Intersection, command);
+      } else {
+        yield _SetStatusCommand(_SetStatus.Difference, command);
+      }
+    }
+  }
+
   /// Given a build commands json file, and the files with local changes,
   /// compute the lint commands to run.
   @visibleForTesting
-  Future<List<Command>> getLintCommandsForChangedFiles(
-    List<dynamic> buildCommandsData,
-    List<io.File> changedFiles,
+  Future<List<Command>> getLintCommandsForFiles(
+    List<Object?> buildCommandsData,
+    List<io.File> files,
+    List<List<Object?>> sharedBuildCommandsData,
+    int? shardId,
   ) async {
-    final List<Command> buildCommands = <Command>[];
-    for (final dynamic data in buildCommandsData) {
-      final Command command = Command.fromMap(data as Map<String, dynamic>);
-      final LintAction lintAction = await command.lintAction;
-      // Short-circuit the expensive containsAny call for the many third_party files.
-      if (lintAction != LintAction.skipThirdParty && command.containsAny(changedFiles)) {
-        buildCommands.add(command);
+    final List<Command> totalCommands = <Command>[];
+    if (sharedBuildCommandsData.isNotEmpty) {
+      final Iterable<Command> buildCommands = buildCommandsData
+          .map((dynamic data) => Command.fromMap(data as Map<String, Object?>));
+      final Iterable<List<Command>> shardBuildCommands =
+          sharedBuildCommandsData.map((List<Object?> list) => list
+              .map((Object? data) =>
+                  Command.fromMap((data as Map<String, Object?>?)!))
+              .toList());
+      final Iterable<_SetStatusCommand> intersectionResults =
+          _calcIntersection(buildCommands, shardBuildCommands);
+      totalCommands.addAll(intersectionResults
+          .where((_SetStatusCommand element) =>
+              element.setStatus == _SetStatus.Difference)
+          .map((_SetStatusCommand e) => e.command));
+      final List<Command> intersection = intersectionResults
+          .where((_SetStatusCommand element) =>
+              element.setStatus == _SetStatus.Intersection)
+          .map((_SetStatusCommand e) => e.command)
+          .toList();
+      // Make sure to sort results, not sure if there is a defined order in the json file.
+      intersection
+          .sort((Command x, Command y) => x.filePath.compareTo(y.filePath));
+      totalCommands.addAll(
+          _takeShard(intersection, shardId!, 1 + shardBuildCommands.length));
+    } else {
+      totalCommands.addAll(buildCommandsData.map((Object? data) => Command.fromMap((data as Map<String, Object?>?)!)));
+    }
+    Stream<Command> filterCommands() async* {
```

Comment:
Please don't add new uses of `Stream`s / `async*`.

Reply:
Are you referring to the guidelines for editing code in the framework?  That doesn't apply here, right?

Reply:
Dart conventions from the framework repo do apply here, yes. But independent of that, `Streams` and `async*` make code very difficult to debug, so should be avoided in general.

Reply:
Done.  I think the framework repo recommendation lists alternatives to streams that don't exist here.  Nevertheless, I removed the `Stream` and the `async*`.